File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    ans = ''
    small = []
    big = []
    for i in range(26):
        small.append(chr(97+i))
        big.append(chr(65+i))
    for i in range(len(s)):
        if s[i] == " ": ans += s[i]
        else:
            # 소문자
            if ord(s[i]) >= 97:
                idx = (small.index(s[i])+n)%26
                ans += small[idx]
            # 대문자
            else:
                idx = (big.index(s[i])+n)%26
                ans += big[idx]
    return ans

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
import java.util.*;
class Solution {
    public String solution(String s, int n) {
        StringBuilder sb = new StringBuilder();
        List<Character> small = new ArrayList<>();
        List<Character> big = new ArrayList<>();
        for (int i=0; i < 26; i++) {
            big.add((char) (65+i)); 
            small.add((char) (97+i));
        }
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (Character.isLowerCase(c)) {
                int idx = (small.indexOf(c) + n) % 26;
                sb.append(small.get(idx));
            } else if (Character.isUpperCase(c)) {
                int idx = (big.indexOf(c) + n) % 26;
                sb.append(big.get(idx));
            } else sb.append(c);
        }
        return sb.toString();
    }
}